src/Http/Routing/src/Matching/DfaMatcherFactory.cs

```diff
@@ -32,9 +32,6 @@ public override Matcher CreateMatcher(EndpointDataSource dataSource)
         // when the services are disposed.
         var lifetime = _services.GetRequiredService<DataSourceDependentMatcher.Lifetime>();
 
-        return new DataSourceDependentMatcher(dataSource, lifetime, () =>
-        {
-            return _services.GetRequiredService<DfaMatcherBuilder>();
-        });
+        return new DataSourceDependentMatcher(dataSource, lifetime, _services.GetRequiredService<DfaMatcherBuilder>);
```

Comment:
Is this right?

Reply:
Yeah, it's `_services.GetRequiredService<DfaMatcherBuilder>` not `_services.GetRequiredService<DfaMatcherBuilder>()`.

Reply:
ahhhh